lib/Doctrine/DBAL/Connection.php

```diff
@@ -814,18 +814,15 @@ public function quoteIdentifier($str)
     /**
      * Quotes a given input parameter.
      *
-     * @param mixed $input The parameter to be quoted.
-     * @param int   $type  The type of the parameter.
+     * @param string $input The parameter to be quoted.
      *
      * @return string The quoted parameter.
      */
-    public function quote($input, $type = ParameterType::STRING) : string
+    public function quote(string $input) : string
     {
         $this->connect();
 
-        [$value, $bindingType] = $this->getBindingInfo($input, $type);
-
-        return $this->_conn->quote($value, $bindingType);
+        return $this->_conn->quote($input);
```

Comment:
`DBAL\Connection` was using `getBindingInfo()` in `quote()`, like it was doing in `resolveParams()`, `executeQuery()` and `executeUpdate()`. But it looks like no code was relying on this "feature", as [no tests are broken](https://travis-ci.org/doctrine/dbal/builds/456856337). `getBindingInfo()` is, AFAICS, only actually used for binding parameters to prepared statements, so this shouldn't be a problem.

Reply:
Looks legit to me. I have an impression that earlier, DBAL unnecessarily combined two ideas: quoting based on the type and mixing binding types with data types. And this is the result.